Core/include/Acts/Utilities/AngleHelpers.hpp

```diff
@@ -29,6 +64,13 @@ Scalar etaFromTheta(Scalar theta) {
 /// @return the polar angle in radian towards the z-axis.
 template <typename Scalar>
 Scalar thetaFromEta(Scalar eta) {
+  if (eta < EtaThetaConversionTraits<Scalar>::minEta) {
+    return std::numbers::pi_v<Scalar>;
+  }
+  if (eta > EtaThetaConversionTraits<Scalar>::maxEta) {
+    return 0;
+  }
+
```

Comment:
```suggestion
```
I don't think these checks do anything, or are necessary. The `thetaFromEta()` function works over the full range.

However, if you want to ensure that the returned &theta; value can be converted back with `etaFromTheta()` to non-`nan`, we would have to try something else.

Reply:
I didn't check for FPEs yet but otherwise it seemed to work yes